search/api_test.py

```diff
@@ -100,192 +95,160 @@ class IndexTests(ESTestCase):
 
-    def test_remove_profile(self):
+    def test_remove_program_enrolled_user(self):
         """
-        Test that remove_profile removes the profile from the index
+        Test that remove_program_enrolled_user removes the user from the index for that program
         """
-        user = UserFactory.create()
-        assert_search(search(), [user])
-        remove_user(user)
-        assert_search(search(), [])
+        program_enrollment = ProgramEnrollmentFactory.create()
+        assert_search(es.search(), [program_enrollment])
+        remove_program_enrolled_user(program_enrollment)
+        assert_search(es.search(), [])
 
-    def test_index_users(self):
+    def test_index_program_enrolled_users(self):
         """
-        Test that index_users indexes an iterable of users
+        Test that index_program_enrolled_users indexes an iterable of program-enrolled users
         """
-        for _ in range(10):
-            with mute_signals(post_save):
-                # using ProfileFactory instead of UserFactory here since UserFactory will not fill in any
-                # fields on Profile
-                profile = ProfileFactory.create()
-            # Not strictly necessary, the muted post_save will prevent indexing
-            remove_user(profile.user)
-
-        # Confirm nothing in index
-        assert_search(search(), [])
-        index_users(User.objects.iterator(), chunk_size=4)
-        assert_search(search(), list(User.objects.all()))
-
-    def test_add_certificate(self):
-        """
-        Test that Certificate is indexed after being added
-        """
-        user = UserFactory.create()
-        assert search()['total'] == 1
-        CachedCertificateFactory.create(user=user)
-        assert_search(search(), [user])
-
-    def test_update_certificate(self):
-        """
-        Test that Certificate is reindexed after being updated
-        """
-        user = UserFactory.create()
-        assert search()['total'] == 1
-        certificate = CachedCertificateFactory.create(user=user)
-        certificate.data = {'new': 'data'}
-        certificate.save()
-        assert_search(search(), [user])
-
-    def test_delete_certificate(self):
-        """
-        Test that Certificate is removed from index after being deleted
-        """
-        user = UserFactory.create()
-        certificate = CachedCertificateFactory.create(user=user)
-        assert_search(search(), [user])
-        certificate.delete()
-        assert_search(search(), [user])
+        with mute_signals(post_save):
+            program_enrollments = [ProgramEnrollmentFactory.build() for _ in range(10)]
+        with patch('search.api._index_program_enrolled_users_chunk', autospec=True, return_value=0) as index_chunk:
+            index_program_enrolled_users(program_enrollments, chunk_size=4)
+            assert index_chunk.call_count == 3
+            index_chunk.assert_any_call(program_enrollments[0:4])
```

Comment:
@noisecapella @giocalitri i refactored this test case (`test_index_program_enrolled_users`, formerly `test_index_users`). this was taking 2.5 sec with calls to ES and to the database, and all it was uniquely testing was that the 'index_' function could handle an iterable. let me know if you have a problem with this refactor